shimmingtoolbox/cli/mask.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+import click
+import nibabel as nib
+import numpy as np
+import os
+
+import shimmingtoolbox.masking.threshold
+from shimmingtoolbox.masking.shapes import shape_square
+from shimmingtoolbox.masking.shapes import shape_cube
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+
+
+@click.group(context_settings=CONTEXT_SETTINGS, help=f"Create a mask based on a specified shape (cube, square, etc.) "
+                                                     f"or based on the thresholding of an input image.")
+def mask_cli():
+    pass
+
+
+@mask_cli.command(context_settings=CONTEXT_SETTINGS,
+              help=f"Create a cube mask from the input file. Return the filename for the output mask.")
+@click.option('-input', 'fname_input', type=click.Path(), required=True, help="Input path of the nifti file to mask")
+@click.option('-output', type=click.Path(), default=os.curdir, help="Output folder for mask in nifti file")
+@click.option("-size", required=True, type=int, help="Length of the side of the cube along first, second and third "
+                                                     "dimension (in pixels)")
+@click.option("-center", nargs=3, type=int, default=(None, None, None), help="Center of the cube along first, second "
+                                                                             "and third dimension (in pixels). If no "
+                                                                             "center is provided, the middle is used.")
+def cube(fname_input, output, size, center):
+    """
+        Create a cube mask from the input file. Return an output nifti file with cube mask.
+
+        Args:
+            fname_input (str): Complete input path of the nifti file to mask.
+            output (str): Output folder for cube mask.
+            size (int): Length of the side of the cube along first, second and third dimension (in pixels).
+            center (int): Center of the cube along first, second and third dimension (in pixels). If no center is
+                            provided, the middle is used.
+            
+        Return:
+            output (str): Filename for the output mask.
+        """
+    # Create the folder where the nifti file will be stored
+    if not os.path.exists(output):
+        os.makedirs(output)
+
+    nii = nib.load(fname_input)
+    data = nii.get_fdata()  # convert nifti file to numpy array
+
+    if len(data.shape) == 3:
+        mask_cb = shape_cube(data, size, size, size, center[0], center[1], center[2])  # creation
```

Comment:
is this working if center is none?

Reply:
The shape_cube function indicates that if the center (for a certain dimension) is None, then the middle of the dimension is used by default. The center is None for all 3 dimensions in my test_cli_mask_cube API and I got no error.